src/main/java/org/opensearch/opnipreprocessing/plugin/OpniPreProcessor.java

```diff
@@ -309,6 +315,26 @@ private void publishToNats (IngestDocument ingestDocument, Connection nc) throws
         nc.publish("raw_logs", payload.toByteArray() );
     }
 
+    private boolean isPendingDelete (IngestDocument ingestDocument, Connection nc) throws Exception {
+        KeyValueManagement kvm = nc.keyValueManagement();
```

Comment:
should we create a management object in every function call or we can create a class-level one and reuse?